demux/demux.c

```diff
@@ -97,11 +97,15 @@ struct demux_opts {
 
 #define OPT_BASE_STRUCT struct demux_opts
 
+#define MAX_BYTES MPMIN(INT64_MAX, (size_t)-1 / 2)
```

Comment:
stdint.h defines `SIZE_MAX`, so no need for the -1 thing

Reply:
`SIZE_MAX - 1` seems to be utilized as the biggest thing to be able to malloc elsewhere, and then bstr.c seems to utilize `size >= SIZE_MAX / 2`, so I guess the thing depends on if how the option max value handling is done (whether > or >=).

If it's > then we still need the -1 methinks.

Reply:
OK, so the type of BYTE_SIZE is int64_t, and `(tmp_int > opt->max)` is how it's checked if I'm reading its int64->long long road correctly (`parse_int64` to `parse_longlong` in `m_option.c`). So we need the -1 as far as I can tell?

Reply:
Switched to `SIZE_MAX / 2` since we compare against INT64_MAX anyways.